enhancements/ingress/add-dns-and-loadbalancer-conditions-to-managed-gateway.md

```diff
@@ -0,0 +1,621 @@
+resources. The "Managed" conditions indicate whether OpenShift should manage the 
+resource, while the "Ready" conditions indicate whether the resource is 
+functioning correctly. This feature is scoped to cloud platform deployments 
+where native LoadBalancer services are available (AWS, Azure, GCP, etc.).
+
+## Motivation
+
+Cluster administrators and end users currently lack visibility into DNS and 
+LoadBalancer provisioning failures for Gateway resources managed by OpenShift. 
+Unlike IngressController resources which provide detailed condition status for 
+DNS and LoadBalancer operations, Gateway resources do not expose this 
+information. This gap makes it difficult to diagnose and troubleshoot issues 
+when Gateways fail due to infrastructure provisioning problems.
+
+By adding these conditions to managed Gateway resources, users gain the same 
+observability they currently have with IngressController, enabling faster issue 
+detection and resolution for DNS and cloud LoadBalancer failures.
+
+### User Stories
+
+* As a customer, I want to know when my Gateway has errors due to a DNS issue 
+so that I can identify and resolve DNS-related problems quickly.
+* As a customer, I want to know when my Gateway has errors due to a LoadBalancer 
+issue so that I can diagnose cloud infrastructure provisioning failures.
+* As a cluster administrator, I want to know when a Gateway has errors due to a 
+DNS issue so that I can proactively address DNS configuration or quota problems.
+* As a cluster administrator, I want to know when my Gateway has errors due to 
+a LoadBalancer issue so that I can troubleshoot cloud provider integration 
+issues or resource limits.
+
+### Goals
+
+* Add `DNSManaged` and `DNSReady` conditions to Gateway status that reflect 
+whether DNS is managed and the state of DNS record provisioning
+* Add `LoadBalancerReady` conditions to Gateway status that reflect whether 
+LoadBalancer is managed and the state of cloud LoadBalancer service provisioning
+* Implement a two-tier condition model: "Managed" conditions indicate whether 
+OpenShift should manage the resource, "Ready" conditions indicate whether it is 
+functioning
+* Ensure conditions follow the same semantics and behavior as existing 
+IngressController DNS and LoadBalancer conditions by reusing the same status 
+computation logic
+* Scope the feature to Gateways in the `openshift-ingress` namespace that are managed
+by Openshift Gateway Class (`openshift.io/gateway-controller/v1`)
+* Support cloud platforms with native LoadBalancer services (AWS, Azure, GCP, etc.)
+
+### Non-Goals
+
+* Supporting bare metal or on-premise deployments that do not have cloud 
+LoadBalancer services
+* Adding these conditions to user-managed Gateway resources outside the 
+`openshift-ingress` namespace
+* Modifying or changing existing IngressController condition behavior or semantics
+* Introducing custom condition types beyond DNS and LoadBalancer at this time
```

Comment:
So this means we don't mark IngressController as Degraded if there are problems with the Gateway?